lib/sycamore/sycamore/writer.py

```diff
@@ -21,8 +26,14 @@ def __init__(self, context: Context, plan: Node):
         self.plan = plan
 
     def opensearch(
```

Comment:
I don't want to break stuff just yet. I feel like having mapwrite.opensearch() vs write.opensearch() is kind of equivalent to having an execute flag but then I have to make sure they're doing the same thing otherwise... 

I feel like if it turns out that the most common use-case is for the write to be nonterminal, then we can just flip the flag default, right? Alternatively maybe some data in the sycamore context object could be used? Will the lineage stuff be explicit in all pipelines?